src/pages/RepoPage/CoverageTab/subroute/FileExplorer/hooks/useRepoBranchContentsTable.tsx

```diff
@@ -0,0 +1,206 @@
+import { CommitErrorTypes } from 'shared/utils/commit'
+import { determineProgressColor } from 'shared/utils/determineProgressColor'
+import CoverageProgress from 'ui/CoverageProgress'
+
+function determineDisplayType(
+  displayType?: string | string[] | ParsedQs | ParsedQs[],
+  isSearching?: boolean
+) {
+  return displayType?.toString().toUpperCase() === displayTypeParameter.list ||
+    isSearching
+    ? displayTypeParameter.list
+    : displayTypeParameter.tree
+}
+
+const defaultQueryParams = {
+  search: '',
+  displayType: '',
+  flags: [],
+  components: [],
+}
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+  path: string
+  branch: string
+}
+
+export function useRepoBranchContentsTable(sortItem?: {
+  ordering?: string
+  direction: string
+}) {
+  const {
+    provider,
+    owner,
+    repo,
+    path: pathParam,
+    branch: branchParam,
+  } = useParams<URLParams>()
+  const { params } = useLocationParams(defaultQueryParams)
+  const { data: repoOverview } = useRepoOverview({
+    provider,
+    repo,
+    owner,
+  })
+
+  const branch = (branchParam || repoOverview?.defaultBranch) as string
+  const location = useLocation()
+
+  const queryParams = qs.parse(location.search, {
+    ignoreQueryPrefix: true,
+    depth: 1,
+  })
+
+  const urlPath = pathParam || ''
+  // useLocationParams needs to be updated to have full types
+  // @ts-expect-error
+  const isSearching = !!params?.search
+  const selectedDisplayType = determineDisplayType(
+    queryParams?.displayType,
+    isSearching
+  )
+
+  const filters = useMemo(() => {
+    return {
+      ...(queryParams?.flags ? { flags: queryParams.flags } : {}),
+      ...(queryParams?.components
+        ? { components: queryParams.components }
+        : {}),
+      displayType: selectedDisplayType,
+      ...(sortItem && {
+        ordering: {
+          direction: sortItem?.direction,
+          parameter: sortItem?.ordering,
+        },
+      }),
+      ...(queryParams?.search && { searchValue: queryParams.search }),
+    }
+  }, [
+    queryParams.flags,
+    queryParams.components,
+    sortItem,
+    queryParams.search,
+    selectedDisplayType,
+  ])
+
+  const { data: branchData, isLoading } = useRepoBranchContents({
+    provider,
+    owner,
+    repo,
+    filters,
+    branch,
+    path: urlPath,
+    opts: {
+      suspense: false,
+    },
+  })
+
+  const indicationRange = branchData?.indicationRange
+
+  const { treePaths } = useTreePaths()
+
+  const finalizedTableRows = useMemo(() => {
+    const createTableData = (branchData: PathContentResultType[]) => {
+      const rawTableRows = branchData.map((result) => {
+        let name
+        if (result?.__typename === 'PathContentDir') {
+          name = (
+            <BranchDirEntry
+              name={result.name}
+              branch={branch}
+              urlPath={urlPath}
+            />
+          )
+        }
+        if (result?.__typename === 'PathContentFile') {
```

Comment:
Can we do an else if here, just to save on the extra computation if the first condition is already meet